com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -546,6 +574,22 @@ def cancel(self, force=False):
             return True
         raise NotImplementedError('Job.cancel()')
 
+    def store_logs(self, filename=None):
+        """Download the console logs and application traces from a job and store them locally in a compressed 'tar' archive (that is, a '.tar.gz' file).
+        
+        Args:
+        filename (str): directory/file path of a compreseed 'tar' archive where the logs and traces should be stored. The default value is 'name.tar.gz' in the current directory, where 'name' is the job name.
+        
+        Returns:
+        str: directory/filename of the compressed 'tar' archive where the logs and traces were stored
+        
+        Exceptions:
+        HTTP GET failed, error xxx xxxxxxxxxxxxx from URL xxxxxxxxxxxxxxxx
+        HTTP GET expected response of type application/x-compressed, got xxxxx/xxxxxx, from URL xxxxxxxxxxxxxx
+        """
+        
+        return self._store_file( ( filename if filename else self.name+'.tar.gz' ), self.applicationLogTrace, 'application/x-compressed')
```

Comment:
The job name contains colons, for example `strings::PyStringWindowTest.tar.gz` would be the output of the logs of the string window test.

The tar command gets confused by the presence of colons while extracting:
```
[streamsadmin@streamsqse]$ tar -xvzf strings::PyStringWindowTest_1000_10_0_app_logs.tar
tar (child): Cannot connect to strings: resolve failed

gzip: stdin: unexpected end of file
tar: Child returned status 128
tar: Error is not recoverable: exiting now
```

In `get_application_logs`, the colons are simply removed to prevent this. I could be convinced that we should keep the colons in the file name, but we should be consistent with our file names at the very least.

Reply:
Good point. I've noted the problem the 'tar' command has with filenames containing colons in the method documentation, and suggested a couple of workarounds.